Doc/c-api/function.rst

```diff
@@ -142,6 +142,16 @@ There are a few functions specific to Python functions.
    do not do type checking. Passing anything other than an instance of
    :c:data:`PyFunction_Type` is undefined behavior.
 
+.. c:function:: PyObject *PyFunction_GET_BUILTINS(PyObject *op)
+
+   Return the dictionary used to look up builtins (which might be ``NULL``)
+   when calling function *op*.
+
+   This function cannot fail and does not do error checking; passing anything
+   other than an instance of :c:data:`PyFunction_Type` is undefined behavior.
+   There is no ``PyFunction_Get*`` equivalent for this function.
```

Comment:
Hm, I don't think C API WG would approve.
Should this function be in a private header (and with an underscore) instead? It looks untested as well.

Reply:
That looks like it's used in a test -- the test assumes that it's working?
Yeah, the more I look into it, the better making it private seems.

cc @ericsnowcurrently, who added `PyFunction_GET_BUILTINS` in #133128